provider/pkg/provider/remote/command.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2016-2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remote
+
+import (
+	"github.com/pulumi/pulumi-go-provider/infer"
+)
+
+type Command struct{}
+
+func (c *Command) Annotate(a infer.Annotator) {
+	a.Describe(&c, `A command to run on a remote host.
+The connection is established via ssh.`)
+}
+
+// The arguments for a remote Command resource
+type CommandInputs struct {
+	// these field annotations allow you to
+	// pulumi:"connection" specifies the name of the field in the schema
+	// pulumi:"optional" specifies that a field is optional. This must be optional.
```

Comment:
I'm not sure what "This must be optional." means. If the type is not optional, it gets a zero value if not filled in.

Reply:
I don't think the type needs to be a pointer. Since all types in go have a zero value, I think it can be any type.